pkg/webhook/resources/upgrade/mutator.go

```diff
@@ -0,0 +1,143 @@
+package upgrade
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	ctlcorev1 "github.com/harvester/harvester/pkg/generated/controllers/core/v1"
+	ctlharvesterv1 "github.com/harvester/harvester/pkg/generated/controllers/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/settings"
+	"github.com/harvester/harvester/pkg/util"
+	"github.com/harvester/harvester/pkg/webhook/types"
+)
+
+func NewMutator(
+	node ctlcorev1.NodeCache,
+	setting ctlharvesterv1.SettingCache,
+) types.Mutator {
+	return &upgradeMutator{
+		node:    node,
+		setting: setting,
+	}
+}
+
+type upgradeMutator struct {
+	types.DefaultMutator
+	node    ctlcorev1.NodeCache
+	setting ctlharvesterv1.SettingCache
+}
+
+func (m *upgradeMutator) Resource() types.Resource {
+	return types.Resource{
+		Names:      []string{"upgrades"},
+		Scope:      admissionregv1.NamespacedScope,
+		APIGroup:   harvesterv1.SchemeGroupVersion.Group,
+		APIVersion: harvesterv1.SchemeGroupVersion.Version,
+		ObjectType: &harvesterv1.Upgrade{},
+		OperationTypes: []admissionregv1.OperationType{
+			admissionregv1.Create,
+		},
+	}
+}
+func (m *upgradeMutator) Create(_ *types.Request, newObj runtime.Object) (types.PatchOps, error) {
```

Comment:
Since we are watching only `create` events, susbsequent additions to the upgrade annotations will have no effects, right?